bookwyrm/forms.py

```diff
@@ -226,6 +227,18 @@ class FileLinkForm(CustomForm):
     class Meta:
         model = models.FileLink
         fields = ["url", "filetype", "availability", "book", "added_by"]
+    
+    def clean(self):
+        """make sure the domain isn't blocked or pending"""
+        cleaned_data = super().clean()
+        url = cleaned_data.get('url')
+        domain = urlparse(url).netloc
+        if models.LinkDomain.objects.filter(domain=domain).exists():
+            status = models.LinkDomain.objects.get(domain=domain).status
+            if status == 'blocked':
+                self.add_error("url", _("Domain is blocked. Don't try this url again."))
```

Comment:
This looks good, I would just tweak the wording a bit so it reads more gently. Maybe something like "This domain is blocked. Please contact your administrator if you think this is an error." ?